pkg/reconciler/kubernetes/tektontrigger/controller.go

```diff
@@ -64,24 +62,60 @@ func NewExtendedController(generator common.ExtensionGenerator) injection.Contro
 			logger.Fatalw("Error creating initial manifest", zap.Error(err))
 		}
 
+		// Reads the source manifest from kodata while initializing the contoller
+		if err := fetchSourceManifests(context.TODO(), &manifest); err != nil {
+			logger.Fatalw("failed to read manifest", err)
+		}
+
+		// Read the release version of pipelines
+		releaseVersion, err := fetchVersion(manifest)
+		if err != nil {
+			logger.Fatalw("failed to read release version from manifest", err)
+		}
+
 		c := &Reconciler{
-			kubeClientSet:     kubeClient,
 			operatorClientSet: operatorclient.Get(ctx),
+			pipelineInformer:  tektonPipelineinformer.Get(ctx),
 			extension:         generator(ctx),
 			manifest:          manifest,
-			pipelineInformer:  tektonPipelineInformer,
+			releaseVersion:    releaseVersion,
 		}
 		impl := tektonTriggerreconciler.NewImpl(ctx, c)
 
-		logger.Info("Setting up event handlers")
+		// Add enqueue func in reconciler
+		c.enqueueAfter = impl.EnqueueAfter
+
+		logger.Info("Setting up event handlers for TektonTrigger")
 
-		tektonTriggersInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+		tektonTriggerinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
 
-		deploymentInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-			FilterFunc: controller.FilterControllerGVK(v1alpha1.SchemeGroupVersion.WithKind("TektonTrigger")),
+		tektonInstallerinformer.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+			FilterFunc: controller.FilterController(&v1alpha1.TektonTrigger{}),
 			Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 		})
 
 		return impl
 	}
 }
+
+// fetchSourceManifests mutates the passed manifest by appending one
+// appropriate for the passed TektonComponent
+func fetchSourceManifests(ctx context.Context, manifest *mf.Manifest) error {
+	var trigger *v1alpha1.TektonTrigger
+	return common.AppendTarget(ctx, manifest, trigger)
+}
+
+func fetchVersion(manifest mf.Manifest) (string, error) {
+	crds := manifest.Filter(mf.CRDs)
+	if len(crds.Resources()) == 0 {
+		return "", fmt.Errorf("failed to find crds to get release version")
+	}
+
+	crd := crds.Resources()[0]
+	version, ok := crd.GetLabels()[releaseLabel]
+	if !ok {
+		return version, fmt.Errorf("failed to find release label on crd")
+	}
+
+	return version, nil
+}
```

Comment:
could we move this logic into the common package and reuse with other components.